### PR TITLE
Fix rootston exit segfaults

### LINKED ISSUE #338: rootston segfault on exit
No time to look at that right now, will do later but that might help me remember. I recall seeing something similar but couldn't find it in open issues.
This happens on exiting with the exit shortcut, on a single rotated screen output / drm backend if that's relevant.

```
0x00007f839c77a3da in wl_signal_emit (signal=0x138, data=0x18) at /usr/include/wayland-server-core.h:387
387		wl_list_for_each_safe(l, next, &signal->listener_list, link)
(gdb) bt
#0  0x00007f839c77a3da in wl_signal_emit (signal=0x138, data=0x18) at /usr/include/wayland-server-core.h:387
#1  0x00007f839c77b616 in wlr_output_destroy (output=0x18) at ../types/wlr_output.c:384
#2  0x00007f839c761bb4 in wlr_drm_backend_destroy (backend=0x564d89c46fe0) at ../backend/drm/backend.c:34
#3  0x00007f839c75f6fe in wlr_backend_destroy (backend=0x564d89c46fe0) at ../backend/backend.c:36
#4  0x00007f839c76a970 in multi_backend_destroy (_backend=0x564d89c46ac0) at ../backend/multi/backend.c:35
#5  0x00007f839c75f6fe in wlr_backend_destroy (backend=0x564d89c46ac0) at ../backend/backend.c:36
#6  0x0000564d89901d08 in main (argc=3, argv=0x7ffeedefb828) at ../rootston/main.c:59
```

### PR DESCRIPTION
**Pull request by martinetd**
Fixes  #338.

### CODE REVIEW

**Comment by SirCmpwn** [comment]
Thanks!

File: backend/libinput/backend.c [hunk 1]
```diff
@@ -99,13 +99,12 @@ static void wlr_libinput_backend_destroy(struct wlr_backend *_backend) {
 	}
 	struct wlr_libinput_backend *backend = (struct wlr_libinput_backend *)_backend;
 	for (size_t i = 0; i < backend->wlr_device_lists->length; i++) {
-		struct wlr_list *wlr_devices = backend->wlr_device_lists->items[i];
-		for (size_t j = 0; j < wlr_devices->length; j++) {
-			struct wlr_input_device *wlr_dev = wlr_devices->items[j];
+		struct wl_list *wlr_devices = backend->wlr_device_lists->items[i];
+		struct wlr_input_device *wlr_dev, *next;
+	        wl_list_for_each_safe(wlr_dev, next, wlr_devices, link) {
```
Comment (emersion) [inline backend/libinput/backend.c]: Tabs !

Comment (SirCmpwn) [reply backend/libinput/backend.c]: Eeek!

Comment (martinetd) [reply backend/libinput/backend.c]: Doh, where did that sneak in from. Repushing something for that, sorry.

**Pull request merged**